pkg/controller/activegate/activegate_controller.go

```diff
@@ -156,6 +165,23 @@ func (r *ReconcileActiveGate) Reconcile(request reconcile.Request) (reconcile.Re
 	return builder.ReconcileAfterFiveMinutes(), nil
 }
 
+func (r *ReconcileActiveGate) reconcilePullSecret(instance dynatracev1alpha1.ActiveGate, log logr.Logger, dtc dtclient.Client) error {
```

Comment:
This should be a pointer.
```suggestion
func (r *ReconcileActiveGate) reconcilePullSecret(instance *dynatracev1alpha1.ActiveGate, log logr.Logger, dtc dtclient.Client) error {
```

Reply:
Done